### PR TITLE
Events for federated secrets controller

### PR DESCRIPTION
**Pull request by mwielgus**
cc: @quinton-hoole @nikhiljindal @kubernetes/sig-cluster-federation

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31705)

### CODE REVIEW

File: federation/pkg/federation-controller/secret/secret_controller.go [hunk 1]
```diff
@@ -290,6 +303,9 @@ func (secretcontroller *SecretController) reconcileSecret(namespace string, secr
 			if !util.ObjectMetaEquivalent(desiredSecret.ObjectMeta, clusterSecret.ObjectMeta) ||
 				!reflect.DeepEqual(desiredSecret.Data, clusterSecret.Data) ||
 				!reflect.DeepEqual(desiredSecret.Type, clusterSecret.Type) {
+
+				secretcontroller.eventRecorder.Eventf(baseSecret, api.EventTypeNormal, "UpdateCluster",
```
Comment (wojtek-t) [inline federation/pkg/federation-controller/secret/secret_controller.go]: UpdateInCluster (consistently with CreateInCluster) ?

File: federation/pkg/federation-controller/secret/secret_controller.go [hunk 2]
```diff
@@ -303,7 +319,12 @@ func (secretcontroller *SecretController) reconcileSecret(namespace string, secr
 		// Everything is in order
 		return
 	}
-	err = secretcontroller.federatedUpdater.Update(operations, secretcontroller.updateTimeout)
+	err = secretcontroller.federatedUpdater.UpdateWithOnError(operations, secretcontroller.updateTimeout,
+		func(op util.FederatedOperation, operror error) {
+			secretcontroller.eventRecorder.Eventf(baseSecret, api.EventTypeNormal, "FailedClusterUpdate",
```
Comment (wojtek-t) [inline federation/pkg/federation-controller/secret/secret_controller.go]: FailedUpdateInCluster ?

File: federation/pkg/federation-controller/secret/secret_controller.go (hunk 1 above)
Comment (mwielgus) [inline federation/pkg/federation-controller/secret/secret_controller.go]: Done.

File: federation/pkg/federation-controller/secret/secret_controller.go (hunk 2 above)
Comment (mwielgus) [inline federation/pkg/federation-controller/secret/secret_controller.go]: Done.

**Comment by mwielgus** [comment]
Agreed with @wojtek-t to add LGTM label once nits resolved.

**Pull request merged**